src/transformers/models/realm/modeling_realm.py

```diff
@@ -0,0 +1,1305 @@
+        )
+        raise
+    tf_path = os.path.abspath(tf_checkpoint_path)
+    logger.info(f"Converting TensorFlow checkpoint from {tf_path}")
+    # Load weights from TF model
+    init_vars = tf.train.list_variables(tf_path)
+    names = []
+    arrays = []
+    for name, shape in init_vars:
+        logger.info(f"Loading TF weight {name} with shape {shape}")
+        array = tf.train.load_variable(tf_path, name)
+        names.append(name)
+        arrays.append(array)
+
+    for name, array in zip(names, arrays):
+
+        # embedder
+        embedder_prefix = "" if isinstance(model, RealmEmbedder) else "embedder/"
+        name = name.replace("module/module/module/bert/", f"{embedder_prefix}bert/")
+        name = name.replace("module/module/LayerNorm/", f"{embedder_prefix}cls/LayerNorm/")
+        name = name.replace("module/module/dense/", f"{embedder_prefix}cls/dense/")
+        name = name.replace("module/module/module/cls/predictions/", f"{embedder_prefix}cls/predictions/")
+
+        name = name.split("/")
+        # adam_v and adam_m are variables used in AdamWeightDecayOptimizer to calculated m and v
+        # which are not required for using pretrained model
+        if any(
+            n in ["adam_v", "adam_m", "AdamWeightDecayOptimizer", "AdamWeightDecayOptimizer_1", "global_step"]
+            for n in name
+        ):
+            logger.info(f"Skipping {'/'.join(name)}")
+            continue
+        pointer = model
+        for m_name in name:
+            if re.fullmatch(r"[A-Za-z]+_\d+", m_name):
+                scope_names = re.split(r"_(\d+)", m_name)
+            else:
+                scope_names = [m_name]
+            if scope_names[0] == "kernel" or scope_names[0] == "gamma":
+                pointer = getattr(pointer, "weight")
+            elif scope_names[0] == "output_bias" or scope_names[0] == "beta":
+                pointer = getattr(pointer, "bias")
+            else:
+                try:
+                    pointer = getattr(pointer, scope_names[0])
+                except AttributeError:
+                    logger.info(f"Skipping {'/'.join(name)}")
+                    continue
+            if len(scope_names) >= 2:
+                num = int(scope_names[1])
+                pointer = pointer[num]
+        if m_name[-11:] == "_embeddings":
+            pointer = getattr(pointer, "weight")
+        elif m_name == "kernel":
+            array = np.transpose(array)
+        try:
+            assert (
+                pointer.shape == array.shape
+            ), f"Pointer shape {pointer.shape} and array shape {array.shape} mismatched"
+        except AssertionError as e:
+            e.args += (pointer.shape, array.shape)
+            raise
+        logger.info(f"Initialize PyTorch weight {name}")
+        pointer.data = torch.from_numpy(array)
+    return model
+
+
+# Copied from transformers.models.bert.modeling_bert.BertEmbeddings
+class BertEmbeddings(nn.Module):
```

Comment:
```suggestion
# Copied from transformers.models.bert.modeling_bert.BertEmbeddings with Bert->Realm
class RealmEmbeddings(nn.Module):
```
Last thing: as you may have seen in the roberta modeling file for instance, we name all classes after the model, so it should be `RealmEmbedding` here and same for all the other classes.

Reply:
Sadly, the `RealmEncoder` model will conflict with BERT `BertEncoder` if I rename `BertEncoder` to `RealmEncoder`. What is your suggestion on this?

Reply:
Ah interesting! Maybe we can leave all the names as Bert then. Wdyt @patrickvonplaten ?

Reply:
Could we maybe rename the "main" `RealmEncoder` to something like `RealmQueryEncoder`?